aspnet/fundamentals/configuration.rst

```diff
@@ -2,169 +2,252 @@
+The following :doc:`controller </mvc/controllers/index>`  uses :doc:`dependency-injection` on :dn:iface:`~Microsoft.Extensions.Options.IOptions\<TOptions>` to access settings:
+ 
+.. literalinclude:: configuration/sample/src/UsingOptions/Controllers/HomeController.cs
   :language: none
+  :start-after: snippet1
+  :end-before: #endregion
+  :dedent: 4
 
-When run, the program will display the default value unless a command line parameter overrides it.
+With the following *appsettings.json* file:
 
-.. image:: configuration/_static/config-console.png
+.. literalinclude:: configuration/sample/src/UsingOptions/appsettings1.json
+  :language: json
 
-.. _options-config-objects:
+The ``HomeController.Index`` method returns ``option1 = value1_from_json, option2 = 2``.
 
-Using Options and configuration objects
----------------------------------------
+In the following code, a second :dn:iface:`~Microsoft.Extensions.Options.IConfigureOptions\<TOptions>` service is added to the service container. It uses a delegate to configure the binding with ``MyOptions``. 
 
-The options pattern enables using custom options classes to represent a group of related settings. A class needs to have a public read-write property for each setting and a constructor that does not take any parameters (e.g. a default constructor) in order to be used as an options class.
+.. literalinclude:: configuration/sample/src/UsingOptions/Startup2.cs
+  :language: c#
+  :start-after: snippet1
+  :end-before: #endregion
+  :dedent: 8
+  :emphasize-lines: 9-13
 
-It's recommended that you create well-factored settings objects that correspond to certain features within your application, thus following the `Interface Segregation Principle (ISP) <http://deviq.com/interface-segregation-principle/>`_ (classes depend only on the configuration settings they use) as well as `Separation of Concerns <http://deviq.com/separation-of-concerns/>`_ (settings for disparate parts of your app are managed separately, and thus are less likely to negatively impact one another).
+There is no limit to the number of ``IConfigureOptions<TOptions>`` services you can add. Each configuration service comes in a NuGet package. They are all applied in order they are registered. Each call to :dn:method:`~Microsoft.Extensions.DependencyInjection.OptionsServiceCollectionExtensions.Configure\<TOptions>` adds an :dn:iface:`~Microsoft.Extensions.Options.IConfigureOptions\<TOptions>` service to the service container. In the example above, the values of ``Option1`` and ``Option2`` are both specified in `appsettings.json`, but the value of ``Option1`` is overridden by the configured delegate in the highlighted code above. When more than one configuration service is enabled, the last configuration source specified “wins”. With the code above, the ``HomeController.Index`` method returns ``option1 = value1_from_action, option2 = 2``.
 
-A simple ``MyOptions`` class is shown here:
+.. note:: Configuration keys are case insensitive.
```

Comment:
Moved it up, removed note. Remove 2 notes, 1 tip and 1 warning all to a bullet list.